Src/GoogleApis.Auth/OAuth2/ClientCredentialParameters.cs

```diff
@@ -0,0 +1,115 @@
+        [Newtonsoft.Json.JsonProperty("private_key")]
+        private string PrivateKey { get; set; }
+
+        /// <summary>Client Email assocaited with ServiceAccountCredential obtained from <a href="https://console.developers.google.com">Google Developers Console</a></summary>
+        [Newtonsoft.Json.JsonProperty("client_email")]
+        public string ClientEmail { get; set; }
+
+        /// <summary>Cliend Id associated with UserCredential created by <a href="https://cloud.google.com/sdk/gcloud/reference/auth/login">gcloud auth login</a>.</summary>
+        [Newtonsoft.Json.JsonProperty("client_id")]
+        private string ClientId { get; set; }
+
+        /// <summary>Client Secret associated with UserCredential created by <a href="https://cloud.google.com/sdk/gcloud/reference/auth/login">gcloud auth login</a>.</summary>
+        [Newtonsoft.Json.JsonProperty("client_secret")]
+        private string ClientSecret { get; set; }
+
+        /// <summary>Refresh Token associated with UserCredential created by <a href="https://cloud.google.com/sdk/gcloud/reference/auth/login">gcloud auth login</a>.</summary>
+        [Newtonsoft.Json.JsonProperty("refresh_token")]
+        public string RefreshToken { get; set; }
+
+        /// <summary>Gets the client secrets which contains the client identifier and client secret. </summary>
+        public ClientCredentialType CredentialType
+        {
+            get
+            {
+                Utilities.ThrowIfNullOrEmpty(Type, "Type");
+
+                switch (Type)
+                {
+                    case "authorized_user":
+                        return ClientCredentialType.UserCredential;
+
+                    case "service_account":
+                        return ClientCredentialType.ServiceAccountCredential;
+
+                    default:
+                        throw new InvalidOperationException(String.Format("Error parsing stream contents. Unrecognized Credential Type {0}.", Type));
+                }
+            }
+        }
+
+        /// <summary>Gets the ClientSecrets portion of the credential parameters.</summary>
+        public ClientSecrets ClientSecrets
+        {
+            get
+            {
+                Utilities.ThrowIfNullOrEmpty(ClientId, "ClientId");
+                Utilities.ThrowIfNullOrEmpty(ClientSecret, "ClientSecret");
+
+                return new ClientSecrets()
+                {
+                    ClientId = this.ClientId,
+                    ClientSecret = this.ClientSecret
+                };
+            }
+        }
+
+        /// <summary>Gets the Base64 unencrypted private key portion of the credential parameters.</summary>
+        public string Base64PrivateKey
+        {
+            get 
+            {
+                Utilities.ThrowIfNullOrEmpty(PrivateKey, "PrivateKey");
+                return PrivateKey.Replace("-----BEGIN PRIVATE KEY-----", "").Replace("\n", "").Replace("-----END PRIVATE KEY-----", "");   
```

Comment:
I wouldn't use getter to perform computation to obtain an intermediate format (not quite PKCS8). Just use a helper method that takes the entire private key (including header and trailer) and converts it to RSAParameters (as described in one of the previous comments).

Also, once you have that helper method, you should probably add a few simple tests (parse wellformed key - get success, parse malformed key - get error)